v-next/hardhat/src/internal/cli/init/project-creation.ts

```diff
@@ -0,0 +1,238 @@
+  }
+
+  return pkg;
+}
+
+async function createEmptyTypescriptHardhatConfig() {
+  await writeEmptyHardhatConfig();
+
+  console.log(`✨ ${chalk.cyan(`Config file created`)} ✨`);
+
+  if (!(await isInstalled(HARDHAT_PACKAGE_NAME))) {
+    console.log("");
+    console.log(`You need to install hardhat locally to use it. Please run:`);
+    const cmd = await getRecommendedDependenciesInstallationCommand({
+      [HARDHAT_PACKAGE_NAME]: `^${await getHardhatVersion()}`,
+    });
+
+    console.log("");
+    console.log(cmd.join(" "));
+    console.log("");
+  }
+
+  console.log();
+
+  showReminderMessages();
+
+  return;
+}
+
+function showReminderMessages() {
+  showStarOnGitHubMessage();
+}
+
+// generated with the "colossal" font
+function printAsciiLogo() {
+  console.log(
+    chalk.blue("888    888                      888 888               888"),
+  );
+  console.log(
+    chalk.blue("888    888                      888 888               888"),
+  );
+  console.log(
+    chalk.blue("888    888                      888 888               888"),
+  );
+  console.log(
+    chalk.blue("8888888888  8888b.  888d888 .d88888 88888b.   8888b.  888888"),
+  );
+  console.log(
+    chalk.blue('888    888     "88b 888P"  d88" 888 888 "88b     "88b 888'),
+  );
+  console.log(
+    chalk.blue("888    888 .d888888 888    888  888 888  888 .d888888 888"),
+  );
+  console.log(
+    chalk.blue("888    888 888  888 888    Y88b 888 888  888 888  888 Y88b."),
+  );
+  console.log(
+    chalk.blue('888    888 "Y888888 888     "Y88888 888  888 "Y888888  "Y888'),
+  );
+  console.log("");
+}
+
+async function printWelcomeMessage() {
+  console.log(
+    chalk.cyan(
+      `👷 Welcome to ${HARDHAT_NAME} v${await getHardhatVersion()} 👷\n`,
+    ),
+  );
+}
+
+async function getAction(): Promise<Action> {
+  // If a matching ENV variable is passed, we do not prompt the user
+  if (
+    process.env.HARDHAT_CREATE_EMPTY_TYPESCRIPT_HARDHAT_CONFIG !== undefined
+  ) {
+    return Action.CREATE_EMPTY_TYPESCRIPT_HARDHAT_CONFIG;
+  }
+
+  // No ENV variable is passed, we prompt the user
+  try {
+    const { default: enquirer } = await import("enquirer");
+
+    const actionResponse = await enquirer.prompt<{ action: string }>([
+      {
+        name: "action",
+        type: "select",
+        message: "What do you want to do?",
+        initial: 0,
+        choices: Object.values(Action).map((a: Action) => {
+          return {
+            name: a,
+            message: a,
+            value: a,
+          };
+        }),
+      },
+    ]);
+
+    const actions: Action[] = Object.values(Action);
+    for (const a of actions) {
+      if (a === actionResponse.action) {
+        return a;
+      }
+    }
+
+    throw new HardhatError(HardhatError.ERRORS.GENERAL.UNSUPPORTED_OPERATION, {
+      operation: `Responding with "${actionResponse.action}" to the project initialization wizard`,
+    });
+  } catch (e) {
+    if (e === "") {
+      return Action.QUIT;
+    }
```

Comment:
Which scenario does this cover? Can we add a default value to the prompt to avoid it?

Reply:
When the user interrupts the cli input request (e.g.: ctrl+c) an error is thrown. We do not want to throw the error, so we just quit

Reply:
Interesting. It's a shame enquirer doesn't provide a more elegant way to handle this. I've seen this pattern a lot in v2. Maybe in the future, we can use a wrapper to abstract it.